File: week3/KJW/스택.py
n = int(input())
stack = []
for _ in range(n):
    command = input().split()
    if command[0] == 'push':
        stack.append(int(command[1]))
    elif command[0] == 'pop':
        if len(stack) == 0:
            print(-1)
        else:
            print(stack.pop())
    elif command[0] == 'size':
        print(len(stack))
    elif command[0] == 'empty':
        if len(stack) == 0:
            print(1)
        else:
            print(0)
    elif command[0] == 'top':
        if stack:
            print(stack[-1])
        else:
            print(-1)

File: week03/ee/스택.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "dev/stdin" : "./input.txt";

const [N, ...arr] = fs.readFileSync(filePath).toString().trim().split("\n");

let stack = [];

for (let i = 0; i < N; i++) {
  // console.log(arr[i])

  const cmd = arr[i].trim().split(" ");
  // console.log(cmd[0])

  switch (cmd[0]) {
    case "push":
        stack.push(cmd[1]);
        break;
    case "pop":
        console.log(stack.length>0 ? stack.pop() : -1);
        break;
    case "size":
        console.log(stack.length);
        break;
    case "empty":
        console.log(stack.length>0 ? 0 : 1);
        break;
    case "top":
        console.log(stack.length>0 ? stack[stack.length-1] : -1);
        break;
    }
}
